node/node.go

```diff
@@ -46,7 +47,9 @@ func NewNode(genDoc *genesis.Genesis, conf *config.Config,
 		"version", version.Version(),
 		"network", genDoc.ChainType())
 
-	network, err := network.NewNetwork(conf.Network)
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
I think we need to add cancel to the node structure and call it when the node is shutting down

Reply:
yes, I can do it, but just tell me what should I do about the network module cancel?

Reply:
I did that but there is an issue, all stuff is fine but I got this warning:
```
the cancel function is not used on all paths (possible context leak)
```
I already called cancel in the stop node method.

Reply:
> I think we need to add cancel to the node structure and call it when the node is shutting down

1. Cancel function just use in parent function, don't pass to other layers.
2. I think don't keep context in structure.

Reply:
> 

Stop method used in any there for cancellation?

Reply:
> I did that but there is an issue, all stuff is fine but I got this warning:
> 
> ```
> the cancel function is not used on all paths (possible context leak)
> ```
> 
> I already called cancel in the stop node method.

`Stop` method use in any there?

Reply:
> I did that but there is an issue, all stuff is fine but I got this warning:
> 
> ```
> the cancel function is not used on all paths (possible context leak)
> ```
> 
> I already called cancel in the stop node method.

Has the Stop method been used anywhere?

Reply:
I don't think so.
where should I use it?

Reply:
in GUI: CTA Exit
in CLI: on interrupt signal

Reply:
ok
I will add it